src/fdb5/database/BaseArchiveVisitor.h

```diff
@@ -46,12 +46,14 @@ class BaseArchiveVisitor : public WriteVisitor {
 
     fdb5::DB* current() const;
 
+protected: // members
+
+    const Key &field_;
```

Comment:
Alignment. Surely "const Key& field_"

field_ has reference type, so & should be part of the type.

Reply:
Where is this being used?

Reply:
The alignment is simply what was there before. This was previous a private member, that Ive promoted to protected so the archive visitor can give it to DB, which passes it to the callback.

Reply:
Note, field_ is a non-canonical key.

Honestly, the treatment of keys is very unclear throughout. You never know when something is canonical, or just a keypart, etc. etc.